trace/kafka/async/option.go

```diff
@@ -0,0 +1,41 @@
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/log"
+)
+
+// OptionFunc definition for configuring the async producer in a functional way.
+type OptionFunc func(*AsyncProducer) error
```

Comment:
Feel free to disregard this comment, but including this commit, Patron would feature these packages:
- trace/kafka/sync
- trace/kafka/async
- async/kafka
- async/kafka/simple
- async/kafka/group

which to me honestly is a little confusing, especially since some of them are using builders, while others still use `OptionFunc`. Would it make sense to at least include a better README description?

Reply:
I wonder why the original PR went down this route? :thinking: 

I did not know about the structure you mention, I will look into improving this.